### PR TITLE
[Azure Firewall IPGroups] - update the API versions and create Bicep

### PR DESCRIPTION
**Pull request by mumian**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@vhorne - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+var jumpBoxNicName = 'JumpHostNic'
+var jumpBoxSubnetId = resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, jumpBoxSubnetName)
+var serverNicName = 'ServerNic'
+var serverSubnetId = resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, serversSubnetName)
+var storageAccountName = '${uniqueString(resourceGroup().id)}sajumpbox'
+var azfwRouteTableName = 'AzfwRouteTable'
+var firewallName = 'firewall1'
+var publicIPNamePrefix = 'publicIP'
+var azureFirewallSubnetId = resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, azureFirewallSubnetName)
+var azureFirewallSubnetJSON = json('{"id": "${azureFirewallSubnetId}"}')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep:50]: why json?

File: quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+    ]
+  }
+}
+var networkSecurityGroupName = '${serversSubnetName}-nsg'
+var azureFirewallIpConfigurations = [for i in range(0, numberOfFirewallPublicIPAddresses): {
+  name: 'IpConf${i}'
+  properties: {
+    subnet: ((i == 0) ? azureFirewallSubnetJSON : json('null'))
+    publicIPAddress: {
+      id: resourceId('Microsoft.Network/publicIPAddresses', '${publicIPNamePrefix}${i + 1}')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep:68]: use the bicep syntax here

File: quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+    }
+  }
+}]
+
+resource ipgroup1 'Microsoft.Network/ipGroups@2021-08-01' = {
+  name: ipgroups_name1
+  location: location
+  properties: {
+    ipAddresses: [
+      '13.73.64.64/26'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep:78]: what ip addresses are these?

File: quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+              destinationPorts: [
+                '90'
+              ]
+            }
+          ]
+        }
+      }
+    ]
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep:426]: move this above properties

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep (hunk 3 above)
Comment (mumian) [reply quickstarts/microsoft.network/azurefirewall-create-with-ipgroups-and-linux-jumpbox/main.bicep:78]: @vhorne - can you answer Brian's question?

**Comment by bmoore-msft** [comment]
pinging @vhorne again...

**Comment by mumian** [comment]
@bmoore-msft - those IP addresses are samples.

**Pull request merged**